cmd/shortener/main.go

```diff
@@ -18,29 +18,35 @@ func main() {
 	cfg.Load()
 	fmt.Println("Running on conf", cfg)
 
-	storage := storage.NewLinksMapping()
-	loadErr := storage.LoadFromJSONFile(cfg.FileStoragePath)
+	db := storage.NewLinksMapping()
+	loadErr := db.LoadFromJSONFile(cfg.FileStoragePath)
 	if loadErr != nil {
 		log.Fatalf("Error encountered on restoring storage from file: %v", loadErr)
 	}
+	app := config.NewApp(cfg, db)
+	if cfg.DBDSN != "" {
+		pool, DBErr := storage.NewDB(cfg.DBDSN)
+		if DBErr != nil {
+			log.Fatalf("Error encountered on connecting to DB: %v", DBErr)
+		}
+		app.DBPool = pool
+		conn, DBErr := pool.Acquire(context.Background())
+		if DBErr != nil {
+			log.Fatalf("Error encountered on acquiring conn: %v", DBErr)
+		}
+		app.Storage = storage.NewDBStorage(conn)
+	}
 
 	r := chi.NewRouter()
 	registerMiddleware(r)
-	registerRoutes(r, storage, cfg)
+	registerRoutes(r, app)
 
 	err := http.ListenAndServe(cfg.ServerAddress, r)
 	if err != nil {
 		log.Fatalf("error ListenAndServe: %v", err)
 	}
 }
 
-func registerRoutes(r *chi.Mux, storage storage.ShortenerStorage, cfg *config.Config) {
-	postShortenHandler := handler.NewPostShortenHandler(storage, cfg)
-	r.Handle("/", postShortenHandler)
-	r.Method(http.MethodPost, "/api/shorten", postShortenHandler)
-	r.Handle("/{id}", handler.NewGetURLHandler(storage, cfg))
-}
-
 func registerMiddleware(r *chi.Mux) {
```

Comment:
It would be better if you move it in `cmd/shortener/routes.go`

Reply:
done